src/haddock/modules/analysis/clustfcc/__init__.py

```diff
@@ -104,10 +104,27 @@ def _run(self):
             self.params['strictness'],
             )
 
-        _, clusters = cluster_fcc.cluster_elements(
-            pool,
-            threshold=self.params['threshold'],
-            )
+        cluster_check = False
+        while not cluster_check:
+            for threshold in range(self.params['threshold'], 0, -1):
+                log.info(f'Clustering with threshold={threshold}')
+                _, clusters = cluster_fcc.cluster_elements(
+                    pool,
+                    threshold=threshold,
+                    )
+                if not clusters:
+                    log.info(
+                        "[WARNING] No cluster was found, increasing threshold!"
```

Comment:
`increasing` or `decreasing` ?

Reply:
Should be `decreasing` since we go from the default of 4 to 3, 2, and 1 (should not go lower than 1!)

Reply:
Yes, Rodrigo implemented it already like that. Just need to update the logging message.

Reply:
ops!

Reply:
> Should be `decreasing` since we go from the default of 4 to 3, 2, and 1 (should not go lower than 1!)

Yes this is how its implemented, just made a typo in the log message.